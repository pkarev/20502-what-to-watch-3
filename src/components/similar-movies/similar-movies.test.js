import React from 'react';
import renderer from 'react-test-renderer';
import SimilarMovies from './similar-movies';

const movies = [
  {
    id: 1,
    name: `Fantastic Beasts`,
    posterSmall: `/img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    posterSmall: `/img/bohemian-rhapsody.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 3,
    name: `Moonrise kindom`,
    posterSmall: `/img/moonrise-kingdom.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 4,
    name: `Moonrise kindom`,
    posterSmall: `/img/moonrise-kingdom.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  }
];

it(`Render SimilarMovies`, () => {
  const tree = renderer
    .create(<SimilarMovies movies={movies} onCardClick={() => {}}/>, {
      createNodeMock: () => ({})
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
