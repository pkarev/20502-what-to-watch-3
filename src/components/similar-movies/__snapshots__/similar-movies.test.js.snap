// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SimilarMovies 1`] = `
<section
  className="catalog catalog--like-this"
>
  <h2
    className="catalog__title"
  >
    More like this
  </h2>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
    >
      <video
        className="small-movie-card__image"
        controls={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        poster="/img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      >
        <source />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Fantastic Beasts
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
    >
      <video
        className="small-movie-card__image"
        controls={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        poster="/img/bohemian-rhapsody.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      >
        <source />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
    >
      <video
        className="small-movie-card__image"
        controls={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        poster="/img/moonrise-kingdom.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      >
        <source />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Moonrise kindom
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
    >
      <video
        className="small-movie-card__image"
        controls={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        poster="/img/moonrise-kingdom.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      >
        <source />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Moonrise kindom
        </a>
      </h3>
    </article>
  </div>
</section>
`;
